### ISSUE TITLE
[P2] Refuse empty evaluation datasets instead of publishing zero-valued reports

### ISSUE THREAD

**Issue by ankaggarwal94** [ankaggarwal94/qanta-buzzer#38]
## Summary

`scripts/evaluate_all.py` can continue after loading zero evaluation questions and publish an `evaluation_report.json` whose all-zero metrics resemble a legitimate 0%-accuracy result, provided a usable sibling training/reference corpus still exists.

At current `main` (`5cafae3c4fc3dfd1525975aa34fda77975b7f4cb`):

- An explicit `--mc-path` to a valid empty JSON list bypasses the default-path fallback because fallback is conditioned on `not args.mc_path`.
- If a nonempty sibling `train_dataset.json` exists, likelihood construction succeeds and the empty evaluation proceeds.
- `summarize_buzz_metrics([])` deliberately returns a fully populated zero-valued metric dict.
- choices-only and plotting helpers largely tolerate empty inputs, so the default reward path can reach report generation without an exception.
- If the train/reference corpus is also absent, a particular likelihood backend such as TF-IDF may crash incidentally; that is not an intentional non-empty evaluation gate and does not protect the common split-artifact case.

This exact failure mode is already recognized and guarded in parts of the T5 split loader, but the main evaluation entrypoint remains permissive.

## Relevant code

- [`scripts/evaluate_all.py`](https://github.com/ankaggarwal94/qanta-buzzer/blob/5cafae3c4fc3dfd1525975aa34fda77975b7f4cb/scripts/evaluate_all.py)
  - Loads the requested dataset, prints `Loaded 0 MC questions`, and lacks a terminal non-empty guard.
- [`evaluation/metrics.py`](https://github.com/ankaggarwal94/qanta-buzzer/blob/5cafae3c4fc3dfd1525975aa34fda77975b7f4cb/evaluation/metrics.py)
  - `summarize_buzz_metrics([])` returns zero for counts/rates/scores.
- [`evaluation/controls.py`](https://github.com/ankaggarwal94/qanta-buzzer/blob/5cafae3c4fc3dfd1525975aa34fda77975b7f4cb/evaluation/controls.py)
  - choices-only returns a zero summary on empty input.

## Minimal reproduction

Start from a normal split-aware artifact directory, then empty only the test split:

```bash
cp -R artifacts/smoke /tmp/qanta-empty-eval
printf '[]\n' > /tmp/qanta-empty-eval/test_dataset.json

# Keep /tmp/qanta-empty-eval/train_dataset.json nonempty.
python scripts/evaluate_all.py \
  --mc-path /tmp/qanta-empty-eval/test_dataset.json \
  --output-dir /tmp/qanta-empty-eval \
  likelihood.model=tfidf
```

### Actual behavior

The script can proceed through the default evaluation/control/report path and emit metrics such as:

```json
{
  "n": 0,
  "buzz_accuracy": 0.0,
  "overall_outcome_accuracy": 0.0,
  "mean_sq": 0.0,
  "ece": 0.0,
  "brier": 0.0
}
```

Those values do not mean the agent scored zero; they mean no agent was evaluated.

### Expected behavior

Evaluation should require at least one valid row and fail before building models, loading stale summaries, running controls, or writing result artifacts. Empty input is an invalid evaluation run, not a valid numerical result.

## Impact

- Dashboards, paper exporters, and comparison scripts can ingest zero-valued results as genuine poor performance.
- CI may pass a pipeline that evaluated nothing.
- A split/filter regression can silently erase test coverage while preserving the artifact shape expected by downstream consumers.
- `ece = 0` and `brier = 0` are especially misleading because they look like perfect calibration/error values rather than missing evidence.

## Proposed resolution

Add a shared non-empty dataset contract to every training/evaluation entrypoint. For evaluation:

```python
if not mc_questions:
    raise ValueError(
        f"Evaluation dataset {mc_path} contains no valid MC questions; refusing to emit metrics"
    )
```

If a machine-readable failure artifact is useful, write a distinct failure receipt/status, never the normal success schema.

## Acceptance criteria

- [ ] Explicit empty `--mc-path` fails nonzero before model construction and report/plot generation, even when a nonempty sibling train split exists.
- [ ] Empty preferred and empty fallback artifacts fail similarly.
- [ ] Malformed inputs that `_safe_load()` converts to empty also fail; corruption is not treated as a 0-row benchmark.
- [ ] No normal `evaluation_report.json`, comparison CSV, or plots are emitted on failure (or temporary outputs are atomically cleaned).
- [ ] Downstream schemas distinguish `n == 0`/missing evidence from measured zero-valued performance.
- [ ] Add equivalent guards to baseline selection, PPO replay, T5 train/val/test evaluation, and control-only entrypoints where empty rows could produce plausible summaries.
- [ ] CI includes a false oracle: replacing only the test data with `[]` while keeping train data valid must fail the pipeline.

## Validation plan

Cover explicit empty list with a valid sibling train corpus, wrapped `{"questions": []}`, malformed JSON, a dataset whose every row is rejected during deserialization/filtering, and a normal one-row dataset. Only the final case should reach metric generation.

**Comment by ankaggarwal94** [comment]
## Independent CoVe revalidation

**Verdict:** CONFIRMED at `main` `5cafae3c4fc3dfd1525975aa34fda77975b7f4cb`  
**Severity:** P2 retained  
**Confidence:** high

The exact empty-input path survives adversarial review:

1. An explicit empty `--mc-path` does not enter the default-path fallback.
2. With a nonempty sibling `train_dataset.json`, TF-IDF construction and prewarming succeed; the evaluation trajectory list is empty.
3. `summarize_buzz_metrics([])`, ECE, Brier, and the choices-only helper all return ordinary zero-valued summaries.
4. `evaluate_all.py` writes `evaluation_report.json` before plot generation. Even if a later plotting function rejects empty arrays, the misleading success artifact has already been emitted.

**Disconfirmation/mitigation found:** if the train/reference corpus is also absent, some backends can fail incidentally before report generation. That is not a deliberate nonempty-evaluation contract and does not protect the normal split-directory reproduction captured in the issue.

The T5 split loader and `compare_policies.evaluate_mlp_policy()` already contain class-adjacent nonempty guards, which supports the issue's claim that protection is inconsistent rather than technically infeasible.

No duplicate issue was found. Issue #23 concerns nonempty targets with an empty MC-construction reference corpus, not an empty evaluation target.

A fresh subprocess reproduction was not run in this connector-only audit; the report-writing path is statically complete and deterministic.
